main.rb

```diff
@@ -0,0 +1,189 @@
+      end
+    end
+    run
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2) [Input the number]:'
+    option = gets.chomp
+    case option
+    when '1' then create_student
+    when '2' then create_teacher
+    else
+      puts 'Please enter a valid number'
+      create_person
+    end
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    print 'Has parent permission: [y/n]'
+    permission = gets.chomp.downcase
+    @people << Student.new(age, name, permission)
+    puts 'Student has been created successfully'
+    run
+  end
+
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    print 'Specialization: '
+    spec = gets.chomp
+    @people << Teacher.new(age, name, spec)
+    puts 'Teacher has been created successfully'
+    run
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+    @books << Book.new(title, author)
+    puts 'Book has been created successfully'
+    run
+  end
+  # rubocop:disable Metrics/MethodLength
+
+  def create_rental
+    if @books.empty?
+      puts 'No books available for rent'
+    elsif @people.empty?
+      puts 'No people available for renting'
+    else
+      puts 'Select a book from the following list by number'
+      @books.each_with_index do |book, index|
+        puts "#{index + 1}. #{book.title} | #{book.author}"
+      end
+      rented_book = gets.chomp.to_i - 1
+      validate_book(rented_book)
+      puts 'Select a person from the following list by number'
+      @people.each_with_index do |person, index|
+        puts "[#{person.type}]  #{index + 1}. Name: #{person.name} | ID: #{person.id} | Age: #{person.age}"
+      end
+      renter = gets.chomp.to_i - 1
+      validate_people(renter)
+      puts 'Enter date'
+      date = gets.chomp
+      new_rental = Rental.new(date, @people[renter], @books[rented_book])
+      @people[renter].add_rental(new_rental)
+      @books[rented_book].add_rental(new_rental)
+      @rentals << new_rental
+      puts 'Rental has been created successfully'
+    end
+    run
+  end
+  # rubocop:enable Metrics/MethodLength
+
+  def validate_book(book)
+    valid_books = (1..@books.length + 1).to_a
+    return unless valid_books.include?(book)
+
+    puts 'Please enter a valid book'
+    create_rental
+  end
+
+  def validate_people(person)
+    valid_people = (1..@people.length + 1).to_a
+    return unless valid_people.include?(person)
+
+    puts 'Please enter a valid user'
+    create_rental
+  end
+
+  def show_rentals
+    if @rentals.empty?
+      puts 'No rentals found'
+    else
+      puts 'Enter ID of Person'
+      renter_id = gets.chomp
+      renter = @people.select { |person| person.id == renter_id }[0]
+      if renter.nil?
+        puts 'ID not found'
+      else
```

Comment:
- [ ] It would be better if after creating the people and book data, and using that data to create a rental, we should be able to list all rentals. However, listing the rentals shows "ID not found" error
![Screenshot from 2022-02-06 03-41-28](https://user-images.githubusercontent.com/46388113/152666919-01ad5661-f464-4f69-ba82-3de51085f42d.png)

Reply:
Hello!

You can see the rentals using the ID listen on the User. So for example, for that person, the ID would be the one highlighted below
![image](https://user-images.githubusercontent.com/59420634/152667128-6e24f74c-94f1-480c-87f9-cb22b962a08d.png)
Do I need to change the logic even if with this id works?

Reply:
Done! I added support for both searches